src/config/server.rs

```diff
@@ -166,11 +160,15 @@ impl Server {
     }
 
     pub fn get_response_headers(&self) -> BTreeMap<String, String> {
-        self.response_headers
-            .clone()
-            .iter()
-            .map(|kv| (kv.key.clone(), kv.value.clone()))
-            .collect()
+        self.headers
+            .as_ref()
+            .map(|h| h.custom.clone())
+            .map_or(BTreeMap::new(), |headers| {
```

Comment:
Use a vector not a BTreeMap. it's totally valid to have two headers with the same name.

Reply:
Are you sure? The old implementation didn't allowed it.